### PR TITLE
Outline improvements

### PR DESCRIPTION
**Pull request by maxtechera**
- Fix outline, prevent pointer-events and fade out outlines
- Make sure the item being outline is the content root
- Refactor Sidebar into Sidekick including ElementHighlighter
- Adds expand path of the hovered item

### CODE REVIEW

File: src/shared/js/components/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ function TreeNode({ id, field, type, displayText, uuid, childNodes, isExpanded,
   };
 
   return (
-    <li className={`csk-sidebar-node ${isExpanded(uuid) ? 'expanded' : 'collapsed'}`}>
+    <li className={`csk-sidebar-node ${isExpanded(uuid) ? 'expanded' : 'collapsed'}`}  onMouseEnter={handleMouseEnter}>
```
Comment (prceasar) [inline src/shared/js/components/Sidebar/Sidebar.js:44]: Why was this moved back to this element? We moved it to the view link because Adam thought it was too disruptive to have this on the top element.

File: src/shared/js/components/Sidebar/Sidebar.js [hunk 2]
```diff
@@ -115,7 +116,7 @@ function Sidebar({ defaultTree }) {
   };
 
   return (
-    <div className="csk-element-sidebar">
+    <div className={`csk-element-sidebar ${show && 'show'}`}>
```
Comment (prceasar) [inline src/shared/js/components/Sidebar/Sidebar.js:119]: This will add a class 'false' to the element

Comment (prceasar) [reply src/shared/js/components/Sidebar/Sidebar.js:119]: when show === false

File: src/shared/js/content.js [hunk 3]
```diff
@@ -10,15 +10,38 @@ import { CSK_ENTRY_ID_NAME, CSK_ENTRY_SELECTOR } from './helpers/constants';
 import { resetBlur, setBlur } from './helpers/blur';
 import getContentfulItemUrl from './helpers/getContentfulItemUrl';
 
+const shrinkContent = () => {
```
Comment (prceasar) [inline src/shared/js/content.js:13]: What is the purpose of the shrink/expand functions?

File: src/shared/js/content.js [hunk 4]
```diff
@@ -121,6 +144,7 @@ const addBlurCode = () => {
   $('body')
     .on('mouseenter', CSK_ENTRY_SELECTOR, handleCskEntryMouseenter)
     .on('mouseleave', CSK_ENTRY_SELECTOR, handleCskEntryMouseleave)
+    .on('mouseover', CSK_ENTRY_SELECTOR, handleCskEntryMouseenter)
```
Comment (prceasar) [inline src/shared/js/content.js:147]: Does it makes sense to have this triggered on both mouseover and mouseenter?

File: src/shared/js/helpers/blur.js [hunk 5]
```diff
@@ -1,4 +1,9 @@
-export const setBlur = ($el, editUrl) => {
+export const setBlur = ($target, editUrl) => {
+  let $el = $target;
+  if (!$el.data('cskEntryId')) {
```
Comment (prceasar) [inline src/shared/js/helpers/blur.js:3]: shouldn't it be 'csk-entry-id'?, I don't think 'cskEntryId' will trigger anything.

File: src/shared/js/helpers/blur.js [hunk 6]
```diff
@@ -1,4 +1,9 @@
...
+    const $parentEl = $target.parents('[data-csk-entry-id]');
```
Comment (prceasar) [inline src/shared/js/helpers/blur.js:4]: What is the purpose of this? Sometimes there are areas which do not have a specific ID, but are there for logical groupings.

File: src/shared/js/components/Sidebar/Sidebar.js (hunk 1 above)
Comment (maxtechera) [reply src/shared/js/components/Sidebar/Sidebar.js:44]: Oh, I was not aware. The UX was weird for me when hovering the list item the page would scroll to the item but I didn't get any highlight.

File: src/shared/js/components/Sidebar/Sidebar.js (hunk 2 above)
Comment (maxtechera) [reply src/shared/js/components/Sidebar/Sidebar.js:119]: I'll update it.

File: src/shared/js/content.js (hunk 3 above)
Comment (maxtechera) [reply src/shared/js/content.js:13]: When showing the sidebar  a `padding-left` is applied to the body to make room for it. If there elements with `position:fixed` those will be hidden behind the sidebar. This functions take care of those elements by adding a custom padding on each of them + whatever the original padding was.

File: src/shared/js/content.js (hunk 4 above)
Comment (maxtechera) [reply src/shared/js/content.js:147]: Maybe not. Without mouseover, I noticed the blur effect wouldn't trigger consistently when moving over items inside another component.

File: src/shared/js/helpers/blur.js (hunk 5 above)
Comment (maxtechera) [reply src/shared/js/helpers/blur.js:3]: Hmm interesting, when accessing with vanilla JS you use `.dataset.cskEntryId` that's where my assumption came from.
And it looks like its working, I'll confirm anyway.

File: src/shared/js/helpers/blur.js (hunk 6 above)
Comment (maxtechera) [reply src/shared/js/helpers/blur.js:4]: The purpose of this is to always trigger the outline for the root of the content item. 
In order to prevent the outline from jumping around the elements that are actually just a field of the parent one.
See my comment here: https://lastrev.atlassian.net/browse/SIDEKICK-16

File: src/shared/js/components/Sidebar/Sidebar.js (hunk 1 above)
Comment (maxtechera) [reply src/shared/js/components/Sidebar/Sidebar.js:44]: Also... it is a bit weird having a button `view` that on hover does the outline but on click behaves like the edit one.

Comment (prceasar) [reply src/shared/js/components/Sidebar/Sidebar.js:44]: Maybe check with Adam on what he thinks

Comment (prceasar) [reply src/shared/js/components/Sidebar/Sidebar.js:44]: Wait.. it should not do anything on click....

File: src/shared/js/content.js (hunk 4 above)
Comment (prceasar) [reply src/shared/js/content.js:147]: ok. As long as it is not behaving strange, I guess it's fine

File: src/shared/js/components/Sidekick/Sidekick.js [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+      const path = getPath({ node: child, uuid });
+      if (path.length) {
+        return [...path, node.uuid];
+      }
+    }
+  }
+  return [];
+};
+
+function Sidebar({ defaultTree }) {
```
Comment (prceasar) [inline src/shared/js/components/Sidekick/Sidekick.js:133]: This can be in another PR, but I would move sidebar out into its own file

**Review by prceasar — APPROVED** [review]

**Review by prceasar — APPROVED** [review]
